consensus/src/lib.rs

```diff
@@ -155,8 +174,48 @@ impl Consensus {
                     (preferred_id, preferred_seq)
                 );
 
+                //adjust batch pool
+                //put back batches from latest proposal if have started deliberation
+                match self.proposals.entry(self.latest_ledger.id) {
+                    Entry::Occupied(mut proposals) => {
+                        match proposals.get_mut().entry(self.node_id) {
+                            Entry::Occupied(proposal) => {
+                                let to_q = proposal.get().proposal.batches.clone();
+                                info!("adjust pool adding {:?} proposal batches", to_q.len());
+                                self.batch_pool.extend(to_q);
```

Comment:
```suggestion
                                info!("adjust pool adding {:?} proposal batches", proposal.get().proposal.batches.len());
                                self.batch_pool.extend(proposal.get().proposal.batches.iter());
```

Reply:
fyi `extend` essentially clones whatever you pass in, so no need to clone the batches before calling `extend`